tools/dotcms-cli/api-data-model/src/main/java/com/dotcms/api/traversal/Filter.java

```diff
@@ -0,0 +1,285 @@
+package com.dotcms.api.traversal;
+
+import static com.dotcms.model.asset.BasicMetadataFields.PATH_META_KEY;
+
+import com.dotcms.model.asset.AssetVersionsView;
+import com.dotcms.model.asset.AssetView;
+import com.dotcms.model.asset.FolderView;
+import java.nio.file.FileSystem;
+import java.nio.file.FileSystems;
+import java.nio.file.PathMatcher;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * A filter that determines whether a path should be included or excluded based on the given
+ * includes and excludes patterns. This filter can be applied to folders and assets separately.
+ */
+public class Filter {
+
+    private final Set<PathMatcher> folderIncludes;
+    private final Set<PathMatcher> assetIncludes;
+    private final Set<PathMatcher> folderExcludes;
+    private final Set<PathMatcher> assetExcludes;
+    private final String rootPath;
+
+    /**
+     * Creates a new filter with the specified includes and excludes patterns for folders and
+     * assets, and the root path.
+     *
+     * @param folderIncludes The patterns to include folders.
+     * @param assetIncludes  The patterns to include assets.
+     * @param folderExcludes The patterns to exclude folders.
+     * @param assetExcludes  The patterns to exclude assets.
+     * @param rootPath       The root path to apply the filter. This is used to remove the root path
+     *                       prefix from the folder and asset paths before applying the filter.
+     */
+    private Filter(
+            Set<PathMatcher> folderIncludes,
+            Set<PathMatcher> assetIncludes,
+            Set<PathMatcher> folderExcludes,
+            Set<PathMatcher> assetExcludes,
+            String rootPath) {
+        this.folderIncludes = folderIncludes;
+        this.assetIncludes = assetIncludes;
+        this.folderExcludes = folderExcludes;
+        this.assetExcludes = assetExcludes;
+        this.rootPath = rootPath;
+    }
+
+    /**
+     * Applies the filter to the given folder, filtering its sub-folders and assets based on the
+     * configured includes and excludes patterns.
+     *
+     * @param folder The folder to apply the filter to.
+     * @return The filtered folder view.
+     */
+    public FolderView apply(FolderView folder) {
+
+        List<FolderView> filteredSubFolders = new ArrayList<>();
+        List<AssetView> filteredAssets = new ArrayList<>();
+
+        if (folder.subFolders() != null) {
```

Comment:
you can use Optional approach, such as
Optional.ofNullable(folder.subFolders()).ifPresent(subfolders -> subfolders.stream().map(this::validateFolder).forEach(filteredSubFolders::add));

Reply:
Done